src/mock_crypto.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright 2018 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// http://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+//! Mock cryptographic primitives.
+//!
+//! These primitives are designed to be very fast (several times faster than the real ones), but
+//! they are NOT secure. They are supposed to be used for testing only.
+
+/// Mock version of a subset of the `rust_sodium` crate.
+pub mod rust_sodium {
+    use rand::{self, FromEntropy, Rng, SeedableRng, XorShiftRng};
+    use std::cell::RefCell;
+
+    thread_local! {
+        static RNG: RefCell<XorShiftRng> = RefCell::new(XorShiftRng::from_entropy());
+    }
+
+    /// Initialise mock `rust_sodium`.
+    pub fn init() -> bool {
+        true
+    }
+
+    /// Initialise mock `rust_sodium` with the given random number generator. This can be used to
+    /// guarantee reproducible test results.
+    pub fn init_with_rng<T: Rng>(other: &mut T) -> Result<(), rand::Error> {
+        let new_rng = XorShiftRng::from_rng(other)?;
+        let _ = RNG.with(move |rng| rng.replace(new_rng));
+        Ok(())
+    }
+
+    /// Mock cryptographic functions.
+    pub mod crypto {
+        /// Mock signing.
+        pub mod sign {
+            use super::super::with_rng;
+            use rand::Rng;
+            use std::ops::{Index, RangeFull};
+
+            /// Number of bytes in a `PublicKey`.
+            pub const PUBLICKEYBYTES: usize = 32;
```

Comment:
Do we need these to be this big ? Can 16 be not sufficient ? So everything in mock is [u8; 16 for e.g. since that is the lowest keccak can output (sha3_128) i think. Or may use a non-crypto but fast hasher like [this one](https://doc.servo.org/fnv/struct.FnvHasher.html) and get a `u64` vai [finish](https://doc.servo.org/core/hash/trait.Hasher.html#tymethod.finish) and then have that in [u8; 8] which will be similar to the `box_` counterparts below and probably faster than tiny-keccak.

These are all suggestions ofc, do only if you don't see a problem with it and can work with seeded-rng to reprocduce failures.

Reply:
It seems that `FnvHasher` is not relevant anymore. According to the [benchmark it refers](https://github.com/Gankro/hash-rs), `DefaultHasher` from the standard lib is much faster. E.g., compare:
```
test fnv::bytes_000002048            ... bench:       1,949 ns/iter (+/- 29) = 1050 MB/s
test sip::bytes_000002048            ... bench:         429 ns/iter (+/- 3) = 4773 MB/s
```
or
```
test fnv::mapcountdense_000002048    ... bench:   2,020,220 ns/iter (+/- 51,216) = 1013 MB/s
test fnv::mapcountsparse_000002048   ... bench:   2,005,899 ns/iter (+/- 43,826) = 1020 MB/s

test sip::mapcountdense_000002048    ... bench:     496,136 ns/iter (+/- 5,623) = 4127 MB/s
test sip::mapcountsparse_000002048   ... bench:     493,424 ns/iter (+/- 4,971) = 4150 MB/s
```

So I'd probably try replacing `tiny_keccak` with the default hasher.